src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -381,6 +405,30 @@ export default class AmalgamatingBusinesses extends Mixins(AmalgamationMixin, Co
     this.isAddingAmalgamatingForeignBusiness = false
   }
 
+  /**
+   * Check if business is already in table and display snackbar text.
+   * @param business The business being added.
+   * @param extraPro Whether we are adding an extra provincial company or not (A companies).
+   */
+  checkForDuplicateInTable (business: any, extraPro: boolean): boolean {
+    // Check if duplicate exists for a company that is not an extra-provincial.
+    const duplicateNonExtraPro = !extraPro &&
```

Comment:
This is the cleanest way possible that came to my mind for doing this.

The nice thing about it is that since the first check here is whether we're checking if extraPro or not, the second part won't execute. This'll save some time and only execute the appropriate find.

Please let me know if you have any suggestions on how to make it even cleaner.

Reply:
I'd be tempted to iterate over the table with a single `find()` that check both identifier (if object is Lear) or corpNumber (if object is Foreign). No need to complicate things.

Reply:
This was causing an error as b.identifier might not be available (if object was foreign). I'll give it another shot (next commit).

Reply:
Yes, you have to check the object type before de-referencing its other values.

Reply:
Example: https://github.com/bcgov/business-create-ui/blob/a7bf48dff97f8cd4b99da488b805b465168da35a/src/mixins/amalgamation-mixin.ts#L53